### PR TITLE
Add back System.Threading types for uapaot build.

### PR DESCRIPTION
**Pull request by AntonLapounov**

### CODE REVIEW

**Comment by AntonLapounov** [comment]
Also remove Helpers.CoreCLR.cs as Jan suggested.

File: src/System.Threading/src/System.Threading.csproj [hunk 1]
```diff
@@ -6,7 +6,14 @@
...
     <IsPartialFacadeAssembly>true</IsPartialFacadeAssembly>
-    <GenFacadesIgnoreMissingTypes Condition="'$(TargetGroup)'=='uapaot'">true</GenFacadesIgnoreMissingTypes>
+  </PropertyGroup>
+  <PropertyGroup Condition="'$(TargetGroup)' == 'uapaot'">
+    <GenFacadesIgnoreMissingTypes>true</GenFacadesIgnoreMissingTypes>
+    <!-- Suppress warning for LockRecursionPolicy type conflicts between this assembly and System.Private.CoreLib. -->
+    <!-- Remove when LockRecursionPolicy is removed from System.Private.CoreLib. -->
+    <NoWarn>0436</NoWarn>
+    <!-- Remove when CDSsyncETWBCLProvider is supported -->
+    <DefineConstants>$(DefineConstants);UAPAOT</DefineConstants>
```
Comment (jkotas) [inline src/System.Threading/src/System.Threading.csproj]: Nit: This should be lowercase (https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/project-guidelines.md at the bottom)

File: src/System.Threading/src/System.Threading.csproj [hunk 2]
```diff
@@ -6,7 +6,14 @@
...
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
     <IsPartialFacadeAssembly>true</IsPartialFacadeAssembly>
-    <GenFacadesIgnoreMissingTypes Condition="'$(TargetGroup)'=='uapaot'">true</GenFacadesIgnoreMissingTypes>
+  </PropertyGroup>
+  <PropertyGroup Condition="'$(TargetGroup)' == 'uapaot'">
+    <GenFacadesIgnoreMissingTypes>true</GenFacadesIgnoreMissingTypes>
+    <!-- Suppress warning for LockRecursionPolicy type conflicts between this assembly and System.Private.CoreLib. -->
+    <!-- Remove when LockRecursionPolicy is removed from System.Private.CoreLib. -->
+    <NoWarn>0436</NoWarn>
+    <!-- Remove when CDSsyncETWBCLProvider is supported -->
```
Comment (jkotas) [inline src/System.Threading/src/System.Threading.csproj]: Could you please open an issue on this and add a link to it here?

File: src/System.Threading/src/System.Threading.csproj (hunk 1 above)
Comment (AntonLapounov) [inline src/System.Threading/src/System.Threading.csproj]: Fixed.

File: src/System.Threading/src/System.Threading.csproj (hunk 2 above)
Comment (AntonLapounov) [inline src/System.Threading/src/System.Threading.csproj]: Done: https://github.com/dotnet/corefx/issues/16881.

**Pull request merged**